src/ott/solvers/nn/losses.py

```diff
@@ -0,0 +1,95 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from types import MappingProxyType
+from typing import Any, Literal, Mapping, Optional, Union
+
+import jax
+import jax.numpy as jnp
+
+from ott.geometry import costs, pointcloud
+from ott.solvers.linear import sinkhorn
+
+
+def monge_gap(
+    samples: jnp.ndarray,
+    mapped_samples: jnp.ndarray,
+    cost_fn: Optional[costs.CostFn] = None,
+    epsilon: Optional[float] = None,
+    relative_epsilon: Optional[bool] = None,
+    scale_cost: Union[bool, int, float, Literal["mean", "max_cost",
+                                                "median"]] = 1.0,
+    sinkhorn_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    **kwargs: Any
+) -> float:
+  r"""Monge gap regularizer :cite:`uscidda:23`.
+
+  For a cost function :math:`c` and an empirical reference :math:`\rho`
+  defined by samples :math:`(x_i)_{i=1,...,n}`, the (entropic) Monge gap
+  of a vector field :math:`T` is defined as:
+  :math:`\mathcal{M}^c_{\rho_x, \epsilon}
+  = \frac{1}{n} \sum_{i=1}^n c(x_i, T(x_i)) - W_\epsilon(\rho, T \# \rho)`.
+  See :cite:`uscidda:23` Eq. (8).
+
+  Args:
+    samples: samples from the reference measure :math:`rho`.
+    mapped_samples: samples from the reference measure :math:`rho`
+    mapped with :math:`T`, i.e. samples from :math:T\sharp\rho.
+    cost_fn: a CostFn function between two points in dimension d.
```

Comment:
CostFn -> cost function

Reply:
Please mention that: If :obj:\`None\`, :class:\`~ott.geometry.costs.SqEuclidean\` is used.